SpatialGDK/Source/SpatialGDK/Public/Schema/MigrationDiagnostic.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Containers/UnrealString.h"
+#include "Engine/EngineBaseTypes.h"
+#include "Engine/GameInstance.h"
+#include "GameFramework/OnlineReplStructs.h"
+#include "Kismet/GameplayStatics.h"
+#include "Schema/Component.h"
+#include "SpatialConstants.h"
+#include "UObject/CoreNet.h"
+#include "Utils/SchemaUtils.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+namespace SpatialGDK
+{
+struct MigrationDiagnostic : Component
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+
+	MigrationDiagnostic() = default;
+
+	static Worker_CommandRequest CreateMigrationDiagnosticRequest()
+	{
+		// Request information from the worker that has authority over the actor that is blocking the hierarchy from migrating
+		Worker_CommandRequest CommandRequest = {};
+		CommandRequest.component_id = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+		CommandRequest.command_index = SpatialConstants::MIGRATION_DIAGNOSTIC_COMMAND_ID;
+		CommandRequest.schema_type = Schema_CreateCommandRequest();
+
+		return CommandRequest;
+	}
+
+	static Worker_CommandResponse CreateMigrationDiagnosticResponse(USpatialNetDriver* NetDriver, Worker_EntityId EntityId,
+																	AActor* BlockingActor)
+	{
+		check(NetDriver != nullptr);
+		check(NetDriver->Connection != nullptr);
+		check(NetDriver->LockingPolicy != nullptr);
+
+		// Respond with information on the worker that has authority over the actor that is blocking the hierarchy from migrating
+		Worker_CommandResponse CommandResponse = {};
+
+		CommandResponse.component_id = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+		CommandResponse.command_index = SpatialConstants::MIGRATION_DIAGNOSTIC_COMMAND_ID;
+		CommandResponse.schema_type = Schema_CreateCommandResponse();
+
+		Schema_Object* ResponseObject = Schema_GetCommandResponseObject(CommandResponse.schema_type);
+		Schema_AddInt64(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID, EntityId);
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID, BlockingActor->GetIsReplicated());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_ID, BlockingActor->HasAuthority());
+		AddStringToSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID, BlockingActor->GetOwner()->GetName());
+
+		AddStringToSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_WORKER_ID, NetDriver->Connection->GetWorkerId());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID, NetDriver->LockingPolicy->IsLocked(BlockingActor));
```

Comment:
I think you can pull  FSpatialLoadBalancingHandler::EvaluateSingleActor in the public section of the class and call it. It will tell us where the local worker thinks the actor should go.

Reply:
Great, done.